src/Coordination/KeeperStateMachine.cpp

```diff
@@ -308,7 +311,25 @@ int KeeperStateMachine::read_logical_snp_obj(
                             s.get_last_log_idx(), latest_snapshot_meta->get_last_log_idx());
             return -1;
         }
-        data_out = nuraft::buffer::clone(*latest_snapshot_buf);
+        int fd = ::open(latest_snapshot_path.c_str(), O_RDONLY);
```

Comment:
Let's move this code into separate function.

Reply:
Done.